File: src/lessons/welcome/SlugHuntingEntity.java
package lessons.welcome;

import java.awt.Color;


public class SlugHuntingEntity extends jlm.universe.bugglequest.SimpleBuggle {

	@Override
	public void run() {
		hunt(); 
	}

	/* BEGIN TEMPLATE */
	public void hunt() {
		// à completer
	/* BEGIN SOLUTION */
		while (! isOverBaggle()) {
			if (isFacingTrail(Color.green)) {
				brushDown();
				forward();
				brushUp();
			} else {
				turnLeft();
			}
		}
		pickUpBaggle();
		/* END SOLUTION */
	}
	/* END TEMPLATE */
	
	boolean isFacingTrail(Color trailColor) {
		if (isFacingWall())
			return false;
		
		forward();
		boolean res = getGroundColor() == trailColor; // equals would be better
		backward();
		return res;
	
	}		
	
	
}

File: src/lessons/welcome/SlugHuntingEntity.py
import java.awt.Color as Color

def isFacingTrail(trailColor):
   if isFacingWall():
      return False
   else:
	  forward()
	  res = getGroundColor().toString() == trailColor.toString()
	  backward()
	  return res

# BEGIN TEMPLATE

def hunt():
   # write your code
   # BEGIN SOLUTION
   while not isOverBaggle():
      brushUp()
      if isFacingTrail(Color.green):
         brushDown()
         forward()
         brushUp()
      else:
	     turnLeft()
   pickUpBaggle()
# END TEMPLATE

hunt()
